### ISSUE TITLE
linter: `import/no-duplicates` should take import attributes into consideration

### ISSUE THREAD

**Issue by maxmilton** [oxc-project/oxc#26871]
### What version of Oxlint are you using?

1.83.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["import"],
  "rules": {
    "import/no-duplicates": "error",
  }
}
```

### What happened?

Inside some of my unit tests I have import code (using [Bun macro import annotation](https://bun.sh/docs/bundler/macros#import-attributes)) like:

```ts
import { compile } from "../src/macro.ts" with { type: "macro" };
import { compile as compileNoMacro } from "../src/macro.ts";
```

This currently reports an error, even though the [import attributes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with) are different. It shouldn't be restricted to just the `"type"` attribute key, any differing import attributes should be treated as a unique import even with the same import path.

I'd like to see a new option added to `import/no-duplicates` (and enabled by default) to differentiate import by their import attributes.

May be worth considering for the `eslint/no-duplicate-imports` rule too.

I couldn't see any existing issue in `eslint-plugin-import` so I create one there too: https://github.com/import-js/eslint-plugin-import/issues/3290

**Comment by maxmilton** [comment]
https://github.com/import-js/eslint-plugin-import/issues/3290 had some interesting discussion, including handling import attributes by default (no rule option).
